Automattic/woocommerce-payments
# Added a filter to disable caching when in a development environment

Fixes #2578

#### Changes proposed in this Pull Request









* Adds a filter which only gets applied in a development or test context which disables report caching (forcing the SQL queries to be made every time the reports are fetched).

#### Testing instructions





Test 1
* Use either a debug log, or a breakpoint to verify that the `return false` statement on line 120 gets hit when you access any of the Analytics pages on the sidebar (e.g. `Analytics -> Overview`.)

Test 2
* Use the dev tools to turn WC Payments dev mode off.
* Go to `/docker/wordpress/wp-config.php` and change the `WP_ENVIRONMENT_TYPE` variable to something other than `development` / `staging` (e.g. `production`).
* Verify the breakpoint no longer gets hit when loading an Analytics page.
* Remember to change your environment type back to development and switch dev mode back on when you're finished! :) 

-------------------


**Post merge**
